src/xcode/ENA/ENA/Source/Scenes/Onboarding/DeltaOnboarding/NewVersionFeatures/DeltaOnboardingNewVersionFeaturesViewController.swift

```diff
@@ -0,0 +1,116 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import UIKit
+
+class DeltaOnboardingNewVersionFeaturesViewController: DynamicTableViewController, DeltaOnboardingViewControllerProtocol, ENANavigationControllerWithFooterChild, UIAdaptivePresentationControllerDelegate {
+
+	// MARK: - Attributes
+
+	var finished: (() -> Void)?
+
+	// MARK: - Initializers
+	
+	init(featureVersion: String = "", hasCloseButton: Bool = true) {
+		self.viewModel = DeltaOnboardingNewVersionFeaturesViewModel()
+		self.hasCloseButton = hasCloseButton
+		super.init(nibName: nil, bundle: nil)
+	}
+
+	@available(*, unavailable)
+	required init?(coder: NSCoder) {
+		fatalError("init(coder:) has not been implemented")
+	}
+
+	// MARK: - View Lifecycle Methods
+
+	override func viewDidLoad() {
+		super.viewDidLoad()
+
+		self.navigationController?.presentationController?.delegate = self
+
+		setupView()
+
+		if hasCloseButton {
+			setupRightBarButtonItem()
+		}
+	}
+
+	// MARK: - Overrides
+
+	override var navigationItem: UINavigationItem {
+		navigationFooterItem
+	}
+
+	private lazy var navigationFooterItem: ENANavigationFooterItem = {
+		let item = ENANavigationFooterItem()
+
+		item.primaryButtonTitle = AppStrings.NewVersionFeatures.buttonContinue
+		item.isPrimaryButtonEnabled = true
+		item.isSecondaryButtonHidden = true
+
+		item.title = AppStrings.NewVersionFeatures.title
+
+		return item
+	}()
+	
+	// MARK: - Protocol UIAdaptivePresentationControllerDelegate
+	
+	func presentationControllerDidDismiss(_ presentationController: UIPresentationController) {
+		finished?()
+	}
+	
+	// MARK: - Protocol ENANavigationControllerWithFooterChild
+
+	func navigationController(_ navigationController: ENANavigationControllerWithFooter, didTapPrimaryButton button: UIButton) {
+		finished?()
+	}
+
+	// MARK: - Private API
+
+	private let viewModel: DeltaOnboardingNewVersionFeaturesViewModel
+	private let hasCloseButton: Bool
+
+	private func setupRightBarButtonItem() {
+		let closeButton = UIButton(type: .custom)
+		closeButton.setImage(UIImage(named: "Icons - Close"), for: .normal)
+		closeButton.setImage(UIImage(named: "Icons - Close - Tap"), for: .highlighted)
+		closeButton.addTarget(self, action: #selector(close), for: .primaryActionTriggered)
+
+		let barButtonItem = UIBarButtonItem(customView: closeButton)
+		barButtonItem.accessibilityLabel = AppStrings.AccessibilityLabel.close
+		barButtonItem.accessibilityIdentifier = AccessibilityIdentifiers.AccessibilityLabel.close
+
+		navigationItem.rightBarButtonItem = barButtonItem
+	}
```

Comment:
Instead you can set the button directly on the item, see newest changes in DeltaOnboardingV15ViewController.swift [here](https://github.com/corona-warn-app/cwa-app-ios/pull/1806/files#diff-887cb279a3db6e6f64863e407a20d19d28d65fd7ca3e5585dd983813e1f38888)

```
item.rightBarButtonItem = CloseBarButtonItem(
 	onTap: { [weak self] in
 		self?.finished?()
 	}
)
```

Reply:
I have changed it...

Reply:
I still see the old code in `setupRightBarButtonItem()`, and I the `func close()` can also be removed. And since you want to set up the close button depending on `hasCloseButton`, maybe you can just move the 
```navigationItem.rightBarButtonItem = CloseBarButtonItem(
 	onTap: { [weak self] in
 		self?.finished?()
 	}
)```
inside the `setupRightBarButtonItem()` instead of the lazy var. Right now it will always show the close button, I think.

Reply:
Hey hey, I have re-worked the code and removed the old lines.

Reply:
Only the rightBarButtonItem snippet inside the navigationItem lazy var is still there (see other comment). Once that is gone, I'm happy 😁

Reply:
I removed it and re-tested it. Should be fine now 😄